### ISSUE TITLE
Latest Version does not allows me to login

### ISSUE THREAD

**Issue by xcrap** [zed-industries/zed#18110]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Since today I can't login in my account.

### Environment

Zed: v0.153.6 (Zed)
OS: macOS 15.0.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

<img width="750" alt="Screenshot 2024-09-20 at 00 43 32" src="https://github.com/user-attachments/assets/b0b3ab31-2238-4f4c-99b1-9c669b2983a6">

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

2024-09-20T00:41:15.510127Z [WARN] unhandled message: ActiveRepo(SupermavenActiveRepoMessage { repo_simple_name: None })
2024-09-20T00:41:33.931418Z [INFO] set status on client 98447: ConnectionError
2024-09-20T00:41:33.948578Z [ERROR] timed out trying to establish connection
2024-09-20T00:41:34.900619Z [WARN] request completed with error: failed to connect to the server
2024-09-20T00:41:34.914957Z [INFO] Opening main db
2024-09-20T00:41:39.698411Z [INFO] set status on client 98447: Reauthenticating
2024-09-20T00:41:39.715738Z [INFO] set status on client 98447: Reconnecting
2024-09-20T00:41:59.768336Z [INFO] set status on client 98447: ConnectionError
2024-09-20T00:41:59.769612Z [ERROR] timed out trying to establish connection

</pre></details>

**Comment by xcrap** [comment]
It's working again.

**Comment by xcrap** [comment]
I'm reopening this, it's happening a lot of times, sometimes he logs in but a lot of times when i open Zed he does not let me sign in and I have a lot of connection errors, maybe some hours later it works, then again the same issue like I'm having just now that I can't even sign in. This was in the latest update I never had this issues before.

**Comment by spenat28** [comment]
It's happening to me all the time, and after couple minutes it does work again ...

MacOS, last version Zed ...

**Comment by spenat28** [comment]
Looks like it's happening a lot. And seems as duplicate issues:

#18117
#16054
#12913
#9578

**Comment by xcrap** [comment]
Yes it keeps happening, sometimes closing and opening the editor works, sometimes not and I have to wait and restart, it's been like this for a few months now.

**Comment by spenat28** [comment]
From what i saw, it may be connected to Zed update, since it does happen to me i think every time i boot Zed and it downloads new version and updates.

Some conflict/not releasing key associated with particular version? Wrong cleanup on destroy()? Something like that might be a culprit, but it's just a guess.

**Comment by shadevkumar** [comment]
Hey @nathansobo i am also facing this same issue, please prioritise this issue .
I was in collaboration mode in chat with one of my friend and suddenly i got logged out of zed and now i am not able to signin.

These are my zed log files

 
```
2024-12-14T13:25:25.963771+05:30 [INFO] set status on client 275717: ConnectionError
2024-12-14T13:25:25.96477+05:30 [ERROR] timed out trying to establish connection
2024-12-14T13:25:30.786451+05:30 [INFO] set status on client 275717: Reauthenticating
2024-12-14T13:25:30.799999+05:30 [INFO] set status on client 275717: Reconnecting
2024-12-14T13:25:50.891864+05:30 [INFO] set status on client 275717: ConnectionError
2024-12-14T13:25:50.892298+05:30 [ERROR] timed out trying to establish connection
```

![image](https://github.com/user-attachments/assets/5687adec-b823-4fda-a710-47e574e08c3e)

Below is my System Environment
```
Zed: v0.165.4 (Zed)
OS: macOS 14.6.0
Memory: 16 GiB
Architecture: aarch64
```

**Comment by notpeter** [comment]
@shadevkumar Do not tag Zed Team members, it's notification spam and will not be tolerated. Consider this a warning.

**Comment by shadevkumar** [comment]
> [@shadevkumar](https://github.com/shadevkumar) Do not tag Zed Team members, it's notification spam and will not be tolerated. Consider this a warning.

Hey,

Thank you for your response. I understand and respect your concerns about notification spam, and I’ll ensure to follow the guidelines moving forward. However, the issue I’ve raised regarding login and connection errors in Zed is critical and directly impacts my ability to collaborate effectively. Here’s why this needs attention:

- The issue persists even after multiple attempts to reconnect.
- I’ve provided logs and system details (as shown in the screenshot) to assist in troubleshooting.
- This is affecting not just me but also other collaborators, as indicated.

Could you please confirm if this issue is being actively addressed or suggest any immediate workarounds? I appreciate the team’s efforts and would value timely communication on this matter.

Thank you for understanding.

Best regards

**Comment by xcrap** [comment]
I stopped having this issue a few updates ago. 
I’m not sure exactly when, but the connection issues did stop. 
I haven’t done anything on my end, so I’m not sure why it got fixed for me. :)
